openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py

```diff
@@ -25,121 +27,64 @@ def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
-        return User.objects.filter(
-            id__in=users_to_be_checked_for_update,
-            courseenrollment__course__in=course_keys,
-            courseenrollment__is_active=True,
-        ).values_list('id', flat=True).distinct()
+        Returns the list of users that have passed the program prerequisite courses
 
-    def check_users_for_application_update(self, users_to_be_checked_for_update, prereq_course_groups):
-        """
-        Check check passed prereq courses for all filtered users
         Args:
-            users_to_be_checked_for_update: List of user ids
-            prereq_course_groups: List of course groups
-        Returns:
-            None
-        """
-        users = User.objects.filter(id__in=users_to_be_checked_for_update)
-        for user in users:
-            self.check_passed_prereq_courses_and_update_application_hub(user, prereq_course_groups)
+            users_to_be_checked_for_update: list of users
 
-    def check_passed_prereq_courses_and_update_application_hub(self, user, prereq_course_groups):
-        """
-        Check passed prereq courses for a particular user and update application hub if all prereq courses are passed
-        Args:
-            user: User object
-            prereq_course_groups: List of course groups
         Returns:
             None
         """
-        for prereq_course_group in prereq_course_groups:
-            if self.is_user_failed_in_course_group(user, prereq_course_group):
-                logger.info('{username} has not yet passed all the pre-reqs'.format(username=user.username))
-                break
-        else:
-            self.update_application_hub(user)
-
-    def is_user_failed_in_course_group(self, user, prereq_course_group):
-        """
-        Checks if user is failed in the given group of courses. User is considered passed if he has passed any one of
-        the courses from course group
-        Args:
-            user: User object
-            prereq_course_group: PrerequisiteCourseGroup model object
-        Returns:
-            boolean, True if the course is failed otherwise False
-        """
-        course_keys = prereq_course_group.open_multilingual_course_keys()
-
-        for course_key in course_keys:
-            course_grade = CourseGradeFactory().read(user, course_key=course_key)
-            course_passed = course_grade and course_grade.passed
+        eligible_users = []
 
-            if course_passed:
-                return False
+        for user in users_to_be_checked_for_update:
+            courses = MultilingualCourseGroup.objects.get_user_program_prereq_courses(user)
+            if has_user_passed_given_courses(user, courses):
+                eligible_users.append(user)
 
-        return True
+        return eligible_users
 
-    def update_application_hub(self, user):
+    def update_is_prerequisite_courses_passed_in_application_hub(self, users):
```

Comment:
If you look at the function `update_is_bu_prerequisite_courses_passed_in_application_hub` from `update_is_bu_prerequisite_courses_passed.py`, it turns out to be exactly same, except the fact that we are updating a different flag in both of the functions. Here is a proposal to reduce this code duplication.

Remove both of these functions from these files and create a new generic helper function:

    def bulk_update_application_hub_flag(flag, users):
        """
        Set the given application hub flag for the provided list of users

        Arguments:
            flag (str): Flag to be updated
            users (list): List of User objects

        Returns:
            None
        """
        user_application_hubs = ApplicationHub.objects.filter(user__in=users)
        user_application_hubs.update(**{flag: True})

        logger.info(f'{flag} flag is updated for all pending users')

This function can then be called in both commands, by passing in the required flag to be updated. If we go with this approach, the unit tests would also have to be updated.